File: ruby/reverse_string.rb
def reverse_string(str)
  # type your code in here
  split_string = str.split("")
  reversed_string = []

  split_string.size.times{reversed_string << split_string.pop}
  
  reversed_string.join("")
  
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
  puts "Expecting: 'enilorac"
  puts "=>", reverse_string('caroline')
end

# Please add your pseudocode to this file
# And a written explanation of your solution

#The first thing I did was that I knew I had to convert the string into an array.
# I did this by using the .split method. 
# Afterwards, I then created an empty array because I wanted to place the rearranged letters into a new array.
# I also assigned a variable split_string to the initial string converted to an array
# Then, in order to reverse the letters, I first had to figure out how many letters were in each array. This was
# accomplished using the .size method
# I then used the .times iterator in conjunction with the push (denoted by the shovel operator <<) and the pop method
# in that for each letter of the split_string array, the .pop method was going to remove the last letter of the array
# and reinsert it into the new (reversed_string) array a set number of times
# the number of times this action was performed was contingent upon the amount of letters in the original array
# after reinserting the letters into the new array, then I used the .join method to turn that array into a string

File: javascript/reverse_string.js
function reverseString(str) {
  // type your code here
   let reversedString = ''

   for (let i = str.length - 1; i >= 0; i--){
     reversedString += str[i]
   }

   return reversedString
   
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));

  console.log("");

  console.log("Expecting: 'enilorac'");
  console.log("=>", reverseString("caroline"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution

// First, I created a new variable and set it equal to an empty string.
//Then, I used a for loop to iterate over each letter in the string starting from the last letter
//and place it into the new string
//Lastly, I return the value of the new string by calling onto it within the function
